### DISCUSSION TITLE
Publishing to the Microsoft Store?

### DISCUSSION THREAD

**Discussion by plettb** [dotnet/maui#18607] [Q&A]
I am having all kinds of trouble publishing my Maui app to the Microsoft Store.

1. If I try publishing from Visual Studio, I get the following error:
- "You must publish one version of this application to the Microsoft Store before using the Store submission automation tool"
2. If I try to upload the package manually to the Microsoft Store (i.e. in a package submission), I get the following errors:
- "Invalid package family name: XXXXXXXX.XXXXXXXXXXXXX_6k51yetae26ce (expected: XXXXXXXX.XXXXXXXXXXXXX_t7g5vnbs3ke9w)" - but I have no control over the suffix that gets added.  Where/how do I change that?
- "Declared package dependency unavailable: Microsoft.WindowsAppRuntime.1.2, Neutral, Universal,Universal" - how do I fix this?  Can I include it as a dependency somehow?

Thanks!

**Comment by VladislavAntonyuk** [comment]
https://youtu.be/EWiksTvfh2w?si=rtNqf9SiPpiITxuw

**Reply by borrmann** [reply]
Hi @VladislavAntonyuk 
I have tried using your approach and saw one of your comments regarding using dotnet publish instead now.

I can successfully create a MSIX with by either using `dotnet publish` or through the `right-click on .csproj, then click publish` method.
However, this does not seem to pick up the Identity from `Package.appxmanifest`, instead, it uses an Identity from a local certificate on the machine. Logically, when I then upload it to partner center, the Identity does not match so the partner center does not allow me to continue.
If I delete the certificate that was used on my machine, the publish command actually fails. 
Do I need to buy or create a certificate, that matches the Id in the partner center? I thought, the whole point of the VS/MS Store publish is, that I don't actually need that?!

I feel like, I must be getting something wrong and was wondering, if you could point me in the right direction..

**Reply by VladislavAntonyuk** [reply]
Hello, @borrmann 
No, you don't need to buy certificates if you publish to Microsoft Store.
You can find my pipeline at VladislavAntonyuk/WorldExplorer. It produces the final msix file that is ready for Microsoft store.

**Comment by robreno** [comment]
I am running into same error when trying to manually upload the .NET Maui MSIX. 

See: https://learn.microsoft.com/en-us/dotnet/maui/windows/deployment/publish-visual-studio-folder?view=net-maui-8.0

See: https://learn.microsoft.com/en-us/windows/apps/publish/publish-your-app/overview?pivots=store-installer-msix

For potential fix will try: https://learn.microsoft.com/en-us/answers/questions/1290018/net-maui-publishing-errors
